forge-ai/src/main/java/forge/ai/ability/PumpAiBase.java

```diff
@@ -166,12 +166,6 @@ public boolean apply(final Card c) {
                 return false;
             }
             return ph.isPlayerTurn(ai) || (combat != null && combat.isAttacking(card) && card.getNetCombatDamage() > 0);
-        } else if (keyword.endsWith("CARDNAME can't be regenerated.")) {
```

Comment:
How should this part be implemented right now? It's a pretty extensive PR, pretty difficult to tell what exactly needs doing in order to make sure the AI doesn't break for a bunch of cards...

Reply:
i probably need to code that in EffectAi, i need to think about it what would be the best logic for this

it probably needs to check if there is already an effect that tires to regenerate the Card
and if the card still can regenerate

Reply:
Alrighty, please assist with this if able :)

Reply:
Effect AI Cases:

* As SubAbility of DealDamage like `Carbonize`, just treat them as bonus
* as Activated Ability with Target from `Furnace Brood`, probably should choose a creature that already has an RegenerationShield or another Ability with `Regeneration$ True` , when using ReplacementList, it already filters if the creature can regenerate or not, just check for Replace Destroy Effects that doesn't use Regeneration, like Totem Armor
* Spell with Target but with Subability like `Gravebind`, but it should just care about the draw one
* Nimbus Cards, like `Knight of the Holy Nimbus`, AI should try to activate it at least once? should check if it would be destroyed by damage or other effects on the stack?

Reply:
i did some logic for the Nimbus cards, for the other cards that uses Target, i just let AI handle the other part